qiita_db/meta_util.py

```diff
@@ -359,38 +359,66 @@ def generate_biom_and_metadata_release(study_status='public'):
+            ms = processing_params.command.merging_scheme
+            software = processing_params.command.software
+            software = '%s v%s' % (software.name, software.version)
 
             # this loop is necessary as in theory an artifact can be
             # generated from multiple prep info files
-            human_cmd = []
+            afps = [fp for _, fp, _ in a.filepaths if fp.endswith('biom')]
+            merging_schemes = []
+            parent_softwares = []
             for p in a.parents:
-                pp = p.processing_parameters
-                # parent is a direct upload; for example per_sample_FASTQ in
-                # shotgun data
-                if pp is None:
-                    pp_cmd_name = ''
+                pparent = p.processing_parameters
+                # if parent is None, then is a direct upload; for example
+                # per_sample_FASTQ in shotgun data
+                if pparent is None:
+                    parent_cmd_name = None
+                    parent_merging_scheme = None
+                    parent_pp = None
+                    parent_software = 'N/A'
                 else:
-                    pp_cmd_name = pp.command.name
-                if pp_cmd_name == 'Trimming':
-                    human_cmd.append('%s @ %s' % (
-                        cmd_name, str(pp.values['length'])))
-                else:
-                    human_cmd.append('%s, %s' % (cmd_name, pp_cmd_name))
-            human_cmd = ', '.join(human_cmd)
+                    parent_cmd_name = pparent.command.name
+                    parent_merging_scheme = pparent.command.merging_scheme
+                    parent_pp = pparent.values
+                    psoftware = pparent.command.software
+                    parent_software = '%s v%s' % (
+                        psoftware.name, psoftware.version)
+
+                merging_schemes.append(qdb.util.human_merging_scheme(
+                    cmd_name, ms, parent_cmd_name, parent_merging_scheme,
+                    processing_params, afps, parent_pp))
+                parent_softwares.append(parent_software)
+            merging_schemes = ', '.join(merging_schemes)
+            parent_softwares = ', '.join(parent_softwares)
 
             for _, fp, fp_type in a.filepaths:
                 if fp_type != 'biom' or 'only-16s' in fp:
                     continue
                 fp = relpath(fp, bdir)
-                # format: (biom_fp, sample_fp, prep_fp, qiita_artifact_id,
-                #          human readable name)
                 for pt in a.prep_templates:
+                    categories = pt.categories()
+                    platform = ''
+                    target_gene = ''
+                    if 'platform' in categories:
+                        platform = ', '.join(
+                            set(pt.get_category('platform').values()))
+                    if 'target_gene' in categories:
+                        target_gene = ', '.join(
```

Comment:
I feel like usually it's better to pass the list of values, instead of concatenating them into a string prematurely.

Reply:
I _think_ this is the right place to concatenate but perhaps I'm missing something, why do you consider that here is premature?

Reply:
My mistake; I reviewed the whole method and there's nothing wrong.